src/Mod/Arch/ArchRebar.py

```diff
@@ -513,7 +514,15 @@ def getIcon(self):
     def setEdit(self, vobj, mode):
 
         if mode == 0:
-            if vobj.RebarShape:
+            if hasattr(vobj.Object, "RebarShape"):
+                try:
+                    # Import module of RebarShape
+                    module = __import__(vobj.Object.RebarShape)
```

Comment:
This seems very sketchy to me. Importing a module in the middle of the viewprovider action. I would rather set this import stuff in the beginning of the file.

Reply:
@vocx-fc I am not sure, because in future there can be `n` number of rebar shapes and if we import all these modules at the beginning of the file, then I consume more time during initializing FreeCAD (I mean at starting).

Python is very intelligent, here if the module already imported and this statement run second-time, then it get cached with real importing (no importing). :)

Reply:
> Python is very intelligent,

Yes, I know this. However, it is still bad style to import inside a function. So, I'm not saying that just adding the imports at the beginning of the file is sufficient. My point is that maybe the entire object needs to be refactored so that it doesn't need tricks like this. For example, the different shapes could be placed in a dictionary that is looked up, or you can delay the import on first use, using LazyLoader class defined in the src/3rdparty, or something like that, I don't know.

I've seen the code of Rebar and of the Reinforcement addon, and personally I don't like it very much, but hey, I'm not going to touch it for the time being.

Reply:
Hi @vocx-fc 

It may not be feasible to import RebarShape module at the beginning of file. Because the RebarShape modules are defined in Reinforcement Addon and that may or may not be installed.
Please suggest if we can still move imports to beginning of file OR suggest the changes to perform.

> My point is that maybe the entire object needs to be refactored so that it doesn't need tricks like this.

The refactoring of object is already done by @berndhahnebach into RebarShape and Rebar Distribution as [rebar2](https://github.com/berndhahnebach/rebar2) addon.
But merging that into FreeCAD without breaking much things is another task.

> I've seen the code of Rebar and of the Reinforcement addon, and personally I don't like it very much, but hey, I'm not going to touch it for the time being.

It will be better if you please suggest the improvements to be done.
I will be happy to follow them while adding new features in Reinforcement Addon.

Thanks

Reply:
> Please suggest if we can still move imports to beginning of file OR suggest the changes to perform.

We could, but I would need to see how everything is handled, and as I said, I haven't looked at the Rebar code recently.

One thing I advise is to check the use of the LazyLoader class to delay the import of the modules until they are actually required. See its usage in the Path Workbench, or in Draft, in `src/Mod/Draft/draftobjects/draftlink.py`. We need `Part` and `DraftGeomUtils`; they are defined at the top, but they are not actually imported then; their import is delayed until the first time they are actually used in the code.

> The refactoring of object is already done by @berndhahnebach into RebarShape and Rebar Distribution as [rebar2](https://github.com/berndhahnebach/rebar2) addon.
> But merging that into FreeCAD without breaking much things is another task.

Yes, I saw Bernd's code, and I like it better, but at the same time, I haven't taken a look at it in months. So, as I said, you don't need to change anything right now; I'm just voicing my opinion.

Reply:
> We could, but I would need to see how everything is handled, and as I said, I haven't looked at the Rebar code recently.

The problem is, the module name depends upon RebarShape name and RebarShapes supported by ReinforcementAddon are fixed for now. But in future, if we need to add more rebar shapes, then we may again need to modify ArchRebar.py to import that RebarShape module.
So, the current approach seems better to support n number of Rebar Shapes.